Shared/Realm.Sync.Shared/SyncConfiguration.cs

```diff
@@ -0,0 +1,55 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Realms.Sync
+{
+    public class SyncConfiguration : RealmConfiguration
```

Comment:
I've added an issue to track API docs - https://github.com/realm/realm-dotnet/issues/929.

My hope is to get this merged soon so that we can tackle the issues, including documentation, in parallel.